independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InterceptorGenerator.java

```diff
@@ -226,7 +227,32 @@ private void addIntercept(MethodCreator intercept, MethodInfo interceptorMethod,
             BranchResult result = intercept.ifNonZero(
                     intercept.invokeVirtualMethod(MethodDescriptors.OBJECT_EQUALS, enumValue, intercept.getMethodParam(0)));
             BytecodeCreator trueBranch = result.trueBranch();
-            Class<?> retType = InterceptionType.AROUND_INVOKE.equals(interceptionType) ? Object.class : void.class;
+            Class<?> retType;
+            if (InterceptionType.AROUND_INVOKE.equals(interceptionType)) {
+                retType = Object.class;
+            } else if (InterceptionType.AROUND_CONSTRUCT.equals(interceptionType)) {
+                if (!interceptorMethod.parameters().equals(
```

Comment:
I'd rather see the validation as a part of an [InterceptorInfo construction](https://github.com/quarkusio/quarkus/blob/master/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InterceptorInfo.java#L55-L81).